products/monitoring/api.yaml

```diff
@@ -964,6 +962,141 @@ objects:
+      A Service-Level Objective (SLO) describes the level of desired good
+      service. It consists of a service-level indicator (SLI), a performance
+      goal, and a period over which the objective is to be evaluated against
+      that goal. The SLO can use SLIs defined in a number of different manners.
+      Typical SLOs might include "99% of requests in each rolling week have
+      latency below 200 milliseconds" or "99.5% of requests in each calendar
+      month return successfully."
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Service Monitoring': 'https://cloud.google.com/monitoring/service-monitoring'
+        'Monitoring API Documentation': 'https://cloud.google.com/monitoring/api/v3/'
+      api: 'https://cloud.google.com/monitoring/api/ref_v3/rest/v3/services.serviceLevelObjectives'
+    parameters:
+    - !ruby/object:Api::Type::String
+      name: service
+      required: true
+      url_param_only: true
+      input: true
+      description: |
+        ID of the service to which this SLO belongs.
+    - !ruby/object:Api::Type::String
+      name: sloId
+      description: |
+        The id to use for this ServiceLevelObjective. If omitted, an id will be generated instead.
+      input: true
+    properties:
+    - !ruby/object:Api::Type::String
+      name: name
+      description: |
+        The full resource name for this service. The syntax is:
+        projects/[PROJECT_ID_OR_NUMBER]/services/[SERVICE_ID]/serviceLevelObjectives/[SLO_NAME]
+      output: true
+    - !ruby/object:Api::Type::String
+      name: displayName
+      description: |
+        Name used for UI elements listing this SLO.
+    - !ruby/object:Api::Type::Double
+      name: goal
+      required: true
+      description: |
+        The fraction of service that must be good in order for this objective
+        to be met. 0 < goal <= 0.999
+    - !ruby/object:Api::Type::String
+      name: rollingPeriod
+      exactly_one_of:
+        - rolling_period
+        - calendar_period
+      description: |
+        A rolling time period, semantically "in the past <rollingPeriod>".
+        Must be an integer multiple of 1 day no larger than 30 days.
+    - !ruby/object:Api::Type::Enum
+      name: calendarPeriod
+      exactly_one_of:
+        - rolling_period
+        - calendar_period
+      description: |
+        A calendar period, semantically "since the start of the current
+        <calendarPeriod>".
+      values:
+        - DAY
+        - WEEK
+        - FORTNIGHT
+        - MONTH
+    - !ruby/object:Api::Type::NestedObject
+      name: serviceLevelIndicator
+      description: |
+        serviceLevelIndicator (SLI) describes a good service.
+        It is used to measure and calculate the quality of the Service's
+        performance with respect to a single aspect of service quality.
+      properties:
+        - !ruby/object:Api::Type::NestedObject
+          name: basicSli
+          exactly_one_of:
```

Comment:
Probably don't need this just with itself

Reply:
Am keeping this mostly to make it easier for splitting my PR up -  i figured that since my next  planned PRs for [requestBased](https://github.com/GoogleCloudPlatform/magic-modules/compare/master...emilymye:sloaddin?expand=1) and [windowBased](https://github.com/GoogleCloudPlatform/magic-modules/compare/master...emilymye:slowindow)) add two more SLIs, there's no harm in using exactly_one_of vs required true

Reply:
It doesn't look like it actually works though- it just ended up with an empty list: https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-3363-old..auto-pr-3363#diff-deebc8e1002911c74cd4f33083bb6d41R152

Reply:
I have seen several PR's now where `exactly_one_of` was used like this. I filed https://github.com/terraform-providers/terraform-provider-google/issues/6103 to try and catch this before it gets to PR's